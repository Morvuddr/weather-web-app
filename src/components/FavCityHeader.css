.favCityHeader {
    width: 90%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin: 0 auto;
}
.deleteButton {
    height: 32px;
    width: 32px;
    border-radius: 50%;
    font-size: 16px;
    line-height: 17.6px;
    background: darkgray;
    color: white;
}