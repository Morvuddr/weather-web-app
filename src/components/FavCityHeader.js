import React from 'react';
import './FavCityHeader.css';

const FavCityHeader = ({ onRemove, id, city }) => (
    <div className='favCityHeader'>
        <h1>{city.name}</h1>
        {!!city.icon && <img src={city.icon} alt='Иконка погоды' />}
        {!!city.temperature && <h1>{city.temperature}</h1>}
        <button className='deleteButton' onClick={() => onRemove(id)}>&#xd7;</button>
    </div>
);

export default FavCityHeader;