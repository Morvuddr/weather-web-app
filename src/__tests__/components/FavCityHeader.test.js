import FavCityHeader from '../../components/FavCityHeader';
import React from 'react';
import renderer from 'react-test-renderer';
describe('FavCityHeader component', () => {

    test('matches the snapshot when city.icon != null, city.temperature != null', () => {
        const city = {
            id: '1',
            name: 'Moscow',
            img: 'abcd',
            temperature: '1'
        };

        const tree = renderer.create(
            <FavCityHeader city={city} onRemove={f => f}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


    test('matches the snapshot when city.icon == null, city.temperature == null', () => {
        const city = {
            id: '1',
            name: 'Moscow',
            img: null,
            temperature: null
        };

        const tree = renderer.create(
            <FavCityHeader city={city} onRemove={f => f}/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});