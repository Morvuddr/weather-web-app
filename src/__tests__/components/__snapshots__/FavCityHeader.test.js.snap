// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavCityHeader component matches the snapshot when city.icon != null, city.temperature != null 1`] = `
<div
  className="favCityHeader"
>
  <h1>
    Moscow
  </h1>
  <img
    alt="Иконка погоды"
    src="abcd"
  />
  <h1>
    1
  </h1>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    ×
  </button>
</div>
`;

exports[`FavCityHeader component matches the snapshot when city.icon == null, city.temperature == null 1`] = `
<div
  className="favCityHeader"
>
  <h1>
    Moscow
  </h1>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    ×
  </button>
</div>
`;
